satellite/marketing/marketingweb/server.go

```diff
@@ -70,31 +102,74 @@ func NewServer(logger *zap.Logger, config Config, listener net.Listener) *Server
 	return s
 }
 
-// ServeHTTP handles index request
-func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+// Serves index page and renders offer and credits tables
+func (s *Server) getOffers(w http.ResponseWriter, req *http.Request){
 	if req.URL.Path != "/" {
 		s.serveNotFound(w, req)
 		return
 	}
 
+	offers, err := s.service.ListAllOffers(context.Background())
+	if err != nil {
+		s.log.Error("app handler error", zap.Error(err))
+		s.serveInternalError(w,req,err)
+		return
+	}
+
 	files := append(s.commonPages(),
 		filepath.Join(s.templateDir, "home.html"),
 		filepath.Join(s.templateDir, "refOffers.html"),
 		filepath.Join(s.templateDir, "freeOffers.html"),
 		filepath.Join(s.templateDir, "roModal.html"),
 		filepath.Join(s.templateDir, "foModal.html"),
 	)
-
-	home, err := template.New("landingPage").ParseFiles(files...)
+	home := template.Must(template.New("landingPage").ParseFiles(files...))
+	err = home.ExecuteTemplate(w, "base", organizeOffers(offers))
 	if err != nil {
-		s.serveInternalError(w, req)
+		s.serveInternalError(w,req,err)
 		return
 	}
+}
 
-	err = home.ExecuteTemplate(w, "base", nil)
+// Helper function used by createOffer to turn post form input into a new offer
+func (s *Server) formToStruct(w http.ResponseWriter, req *http.Request) (o marketing.NewOffer, e error){
```

Comment:
Does this function need to have the receiver here?

Reply:
Since I'm not using anything from `s` no it doesn't.

Reply:
done